src/deferred.js

```diff
@@ -183,8 +186,13 @@ jQuery.extend( {
 								},
 
 								// Only normal processors (resolve) catch and reject exceptions
+								// Exceptions in other cases should not effect the promise result
 								process = special ?
-									mightThrow :
+									function() {
+										try {
+											mightThrow();
+										} catch ( e ) {}
```

Comment:
Swallowing errors from progress callbacks is a problem. We just went through a big exercise with `ready` to prevent its analogous situation.